internal/services/servicebus/servicebus_namespace_resource.go

```diff
@@ -543,7 +543,11 @@ func flattenServiceBusNamespaceEncryption(encryption *namespaces.Encryption) ([]
 		}
 		keyId = keyVaultKeyId.ID()
 		if props.Identity != nil && props.Identity.UserAssignedIdentity != nil {
-			identityId = *props.Identity.UserAssignedIdentity
+			sbnUaiId, err := commonids.ParseUserAssignedIdentityIDInsensitively(*props.Identity.UserAssignedIdentity)
+			if err != nil {
+				return nil, fmt.Errorf("parsing %q as a User Assigned Identity ID: %+v", *props.Identity.UserAssignedIdentity, err)
```

Comment:
The resource ID parsing functions already return error messages with detailed information on the string that it's attempting to parse and as what ID type, so there is no need to wrap these error messages further with redundant information
```suggestion
				return nil, err
```